### ISSUE TITLE
Bug in exists method?

### ISSUE THREAD

**Issue by ElLeopard** [mchidk/BinaryRage#23]
This code doesn't work for me unless I comment out the WaitForCompletion method.
Maybe it's a race condition, the data is not (yet) written to disk, and the exists method only checks
on disk?

BinaryRage.DB.Insert(key, document, StoreDirectory);
//BinaryRage.DB.WaitForCompletion();
bool exists = BinaryRage.DB.Exists(key, StoreDirectory);

Console.WriteLine(exists); // False!

**Comment by tormaroe** [comment]
Yes, this is true. For the same reason `BinaryRage.DB.Remove` will fail if it hasn't reached the disk yet. The user should not have to care about there being a cache, and that persistance is async. Leaky abstraction. I believe this should be fixed.
